### PR TITLE
feat: Adding JSON schema for alibaba source plugin

### PR DESCRIPTION
**Pull request by mnorbury**
- Add `gen-spec-schema` to Makfile
- Adding spec generation code
- Adding test cases
- Configure plugin to use JSON schema

fixes: https://github.com/cloudquery/cloudquery/issues/16334

### CODE REVIEW

File: plugins/source/alicloud/client/spec.go [hunk 1]
```diff
@@ -49,3 +54,38 @@ func (s *Spec) Validate() error {
...
+	sc.Properties.Set("accounts", &accounts)
+
+	sc.Required = append(sc.Required, "accounts")
+}
+
+func (AccountSpec) JSONSchemaExtend(sc *jsonschema.Schema) {
+	one := uint64(1)
+
+	name := *sc.Properties.Value("name")
+	name.MinLength = &one
```
Comment (candiduslynx) [inline plugins/source/alicloud/client/spec.go:72]: use annotation instead
```go
type AccountSpec struct {
	Name      string   `json:"name,omitempty" jsonschema:"required,minLength=1"`
	Regions   []string `json:"regions,omitempty" jsonschema:"required,minItems=1"`
	AccessKey string   `json:"access_key,omitempty" jsonschema:"required,minLength=1"`
	SecretKey string   `json:"secret_key,omitempty" jsonschema:"required,minLength=1"`
}
```

File: plugins/source/alicloud/client/spec.go [hunk 2]
```diff
@@ -49,3 +54,38 @@ func (s *Spec) Validate() error {
...
 }
+
+func (Spec) JSONSchemaExtend(sc *jsonschema.Schema) {
+	one := uint64(1)
+
+	accounts := *sc.Properties.Value("accounts").OneOf[0]
+	accounts.MinItems = &one
+	sc.Properties.Set("accounts", &accounts)
+
+	sc.Required = append(sc.Required, "accounts")
```
Comment (candiduslynx) [inline plugins/source/alicloud/client/spec.go:65]: add to annotation instead of extend

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/source/alicloud/client/spec.go (hunk 1 above)
Comment (mnorbury) [reply plugins/source/alicloud/client/spec.go]: Addressed in https://github.com/cloudquery/cloudquery/pull/16355/commits/baa62595a3f268025cc062b4d5900fe069e4e3bd

File: plugins/source/alicloud/client/spec.go (hunk 2 above)
Comment (mnorbury) [reply plugins/source/alicloud/client/spec.go]: Addressed in https://github.com/cloudquery/cloudquery/pull/16355/commits/baa62595a3f268025cc062b4d5900fe069e4e3bd

**Review by hermanschaaf — APPROVED** [review]

File: plugins/source/alicloud/client/spec/gen/main.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	cqjsonschema "github.com/cloudquery/codegen/jsonschema"
+	"github.com/invopop/jsonschema"
+)
+
+func main() {
+	fmt.Println("Generating JSON schema for plugin spec")
+	cqjsonschema.GenerateIntoFile(new(client.Spec), path.Join(currDir(), "../..", "schema.json"),
+		cqjsonschema.WithAddGoComments("github.com/cloudquery/cloudquery/plugins/source/alicloud/client", path.Join(currDir(), "../..")),
+		func(reflector *jsonschema.Reflector) {
+			reflector.NullableFromType = false
```
Comment (candiduslynx) [inline plugins/source/alicloud/client/spec/gen/main.go:19]: yay!

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**